csharp/src/Apache.Arrow/Schema.cs

```diff
@@ -63,5 +63,49 @@ public int GetFieldIndex(string name, StringComparer comparer = default)
             return _fields.IndexOf(
                 _fields.Single(x => comparer.Compare(x.Name, name) == 0));
         }
+
+        public void RemoveField(int fieldIndex)
+        {
+            if (fieldIndex < 0 || fieldIndex > _fields.Count)
+            {
+                throw new ArgumentException(nameof(fieldIndex), "Invalid fieldIndex");
+            }
+            _fieldsDictionary.Remove(_fields[fieldIndex].Name);
+            _fields.RemoveAt(fieldIndex);
+        }
+
+        public void AddField(Field newField, int fieldIndex)
```

Comment:
In .NET, this operation is typically called `Insert`.

See [https://docs.microsoft.com/en-us/dotnet/api/system.collections.ilist.insert](https://docs.microsoft.com/en-us/dotnet/api/system.collections.ilist.insert)